### PR TITLE
Feature/dockerfiles

### LINKED ISSUE #114: create docker containers for saps components
depends on #63 
is related to #60

### PR DESCRIPTION
**Pull request by wesleymonte**
Resolves #114

### CODE REVIEW

File: docker/dockerfiles/catalog [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM postgres:10
+
+LABEL maintainer="wesley.monte@ccc.ufcg.edu.br"
+
+ENV POSTGRES_USER=admin
+ENV POSTGRES_PASSWORD=admin
+ENV POSTGRES_DB=saps
```
Comment (raonismaneoto) [inline docker/dockerfiles/catalog:7]: these data are sensitive. Maybe you should create a script that reads the data from a conf file and fulfill fields/create env variables wherever this data is needed. One option is: You can pass the variables as parameters when u run the container...

File: docker/dockerfiles/dispatcher [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    python-shapely
+
+WORKDIR /dispatcher
+
+COPY ./target/saps-engine-0.0.1-SNAPSHOT.jar dispatcher.jar
+COPY ./target/lib  lib/
+COPY ./resources resources/
+COPY ./scripts  scripts/
+
+ENTRYPOINT java -Dlog4j.configuration=file:"${LOG_PROPERTIES_PATH}" -cp dispatcher.jar:lib/* "${MAIN_CLASS}" "${CONFIG_PATH}"
```
Comment (raonismaneoto) [inline docker/dockerfiles/dispatcher:34]: just to confirm, what exactly the -cp part do?

File: docker/stack.sh [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+# Script useful for manage stack saps services life-cycle.
+# See usage for commands & options.
+
+set -o errexit;
+
+readonly ARCHIVER_REPO=wesleymonte/archiver
+readonly CATALOG_REPO=wesleymonte/catalog
+readonly DISPATCHER_REPO=wesleymonte/dispatcher
+readonly SCHEDULER_REPO=wesleymonte/scheduler
```
Comment (raonismaneoto) [inline docker/stack.sh]: In the future you guys should create a saps docker hub, so you could keep that containers there. @thiagomanel

File: docker/stack.sh [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+readonly ARCHIVER_CONTAINER=saps-archiver
+readonly CATALOG_CONTAINER=saps-catalog
+readonly DISPATCHER_CONTAINER=saps-dispatcher
+readonly SCHEDULER_CONTAINER=saps-scheduler
+
+readonly CATALOG_USER=admin
+readonly CATALOG_PASSWORD=admin
+readonly CATALOG_DB=saps
```
Comment (raonismaneoto) [inline docker/stack.sh:21]: same issue as above. You should have an env-composer script that fulfill this fields by reading the sensitive data from a conf file.

File: docker/stack.sh [hunk 5]
```diff
@@ -0,0 +1,227 @@
+#!/bin/bash
```
Comment (raonismaneoto) [inline docker/stack.sh:1]: what a beautiful code! Congrats!

**Review by raonismaneoto — CHANGES_REQUESTED** [review]

File: docker/stack.sh [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+      docker push "${DISPATCHER_REPO}":"${tag}"
+      ;;
+    scheduler)
+      docker push "${SCHEDULER_REPO}":"${tag}"
+      ;;
+  esac
+}
+
+define_params() {
+  case $1 in
```
Comment (raonismaneoto) [inline docker/stack.sh:226]: just another suggestion. To make this script more unix-like, you could use getopts and when you call the script you should call like ./stack.sh -p access [other params]. If you have time, I strongly recommend that.

File: docker/dockerfiles/catalog (hunk 1 above)
Comment (thiagomanel) [reply docker/dockerfiles/catalog:7]: not sure about that. But, we can replace them by placeholders, e.g. $POSTGRES_USER, anyway.

File: docker/stack.sh (hunk 3 above)
Comment (thiagomanel) [reply docker/stack.sh]: how about now?

File: docker/stack.sh (hunk 4 above)
Comment (thiagomanel) [reply docker/stack.sh:21]: not sure

File: docker/dockerfiles/dispatcher (hunk 2 above)
Comment (thiagomanel) [reply docker/dockerfiles/dispatcher:34]: I think it means classpath

File: docker/stack.sh (hunk 6 above)
Comment (thiagomanel) [reply docker/stack.sh:226]: I'm not against this suggestion ... but, you guys should keep in mind that is hard to make things simple, so beware of not overkilling this new incarnation of saps.

**Review by thiagomanel — CHANGES_REQUESTED** [review]
almost done

File: docker/dockerfiles/dispatcher (hunk 2 above)
Comment (wesleymonte) [reply docker/dockerfiles/dispatcher:34]: Exact.

File: docker/stack.sh (hunk 4 above)
Comment (wesleymonte) [reply docker/stack.sh:21]: Believe that it is unnecessary, this script is aimed at the development environment. It is easier to just change the value of the variables.

Comment (wesleymonte) [reply docker/stack.sh:21]: Believe that it is unnecessary, this script is aimed at the development environment. It is easier to just change the value of the variables.

File: docker/dockerfiles/dispatcher (hunk 2 above)
Comment (wesleymonte) [reply docker/dockerfiles/dispatcher:34]: Exact.

File: docker/dockerfiles/catalog (hunk 1 above)
Comment (wesleymonte) [reply docker/dockerfiles/catalog:7]: These variables exist by default in postgres. Assigning them in this way, define a default user for the catalog. They can be changed at container startup time.

File: docker/stack.sh (hunk 5 above)
Comment (wesleymonte) [reply docker/stack.sh:1]: Thanks @raonismaneoto!

**Review by raonismaneoto — APPROVED** [review]

File: docker/stack.sh (hunk 3 above)
Comment (wesleymonte) [reply docker/stack.sh]: Done. [Dockerhub](https://hub.docker.com/orgs/ufcgsaps/repositories).

**Review by thiagomanel — APPROVED** [review]
thanks @wesleymonte

**Pull request merged**